### ISSUE TITLE
On Dark theme "X" close tab button in tabs list has low color contrast against white pages

### ISSUE THREAD

**Issue by anna-yeddi** [mozilla-mobile/firefox-ios#27222]
## Steps to reproduce
1. Have Dark system theme on
2. Open a tab and navigate to a page with a solid white background, i.e. [this example](https://cssbud.com/code-templates/blank-html-external-css-and-js/)
3. Navigate to and activate "Tabs open" button on the toolbar
4. Locate and review the "X" (Close tab) button for the tab from step 2

### Expected behavior
1. Close button is clearly visible against all web page backgrounds, incl. solid white. The color contrast of the button's background and its `x` icon should be at least 3:1

### Actual behavior
1. Color contrast of the `x` (Close tab) button appears to have color contrast of 2.1:1 when viewed against white background of the page

### Device & build information
* Device: iPhone 15 Pro, iOS version 18.5
* Build version: Firefox 139.1 (56119)
* First seen version: Firefox 139.1 (56119)
### Notes
Attachments: 
<img width="1012" alt="Colour Contrast Analyzer output with 2.1:1 color contrast for X buttons light gray against white web page background" src="https://github.com/user-attachments/assets/b7c54f13-a044-4f21-a6a1-cd229b8ce16c" />

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-12487)
